### PR TITLE
Add scoped admins managed within Decidim Awesome

### PR DESCRIPTION
**Pull request by microstudi**
This PR (will) allow to add any user as a "fake" admin and scope it using the same constrains as custom styles for instance.
This admins will differ a little from current space admins as they won't receive any notification for instance.

- [x] scoped admins config controller && commands
- [x] select2 user picker
- [x] working concept
- [ ] ~~hide non-accessible admin menus~~
- [x] allow access to index `/admin`
- [x] allow access to index participatory spaces if slug constraint
- [ ] allow access to index participatory spaces && component index if component constraint
- [x] redirect when accessing  forbidden places
- [x] set users as admin outsite admin
- [x] scope to "never" by default when adding a box
- [ ] ~Do not allow unscoped admins~
- [x] Do not allow scoped admins administrate awesome (or any other type of sub-space admin) fixes #admin
- [x] request analyzer && model tamperer middleware 

tests
- [x] command tests
- [ ] admin controller tests (select2)
- [x] middleware tests
- [x] system tests

### Development note:

Due the way Rails handles the 404 error it is necessary to deactivate exception capture to test redirections on forbidden paths (this is the way it works on production). To configure development this way, edit `config/environtments/developement.rb` and change the following line:

```ruby
  # Show full error reports.
  config.consider_all_requests_local = false
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Usability and UX tweaks for Decidim.
 This plugin allows the administrators to expand the possibilities of Decidim beyond some existing limitations.
 All tweaks are provided in a optional fashion with granular permissions that let the administrator to choose exactly where to apply those mods. Some tweaks can be applied to any assembly, other in an specific participatory process or even in type of component only.
 
-**This in beta status, we do not accept any responsibility for breaking anything. Feedback is appreciated though.**
+**DISCLAIMER: This module is heavily tested and widely used, howevever we do not accept any responsibility for breaking anything. Feedback is appreciated though.**
```
Comment (verarojman) [inline README.md:13]: 👏🏾 ❤️  Awesome is awesome

File: app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+module Decidim
+  module DecidimAwesome
+    module Admin
+      class CreateScopedAdmin < Rectify::Command
+        # Public: Initializes the command.
+        #
+        def initialize(organization)
+          @organization = organization
+          @ident = rand(36**8).to_s(36)
```
Comment (verarojman) [inline app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb:11]: Why aren't you using the user's id?

**Comment by verarojman** [comment]
I'm getting an error when:

- Logging in with user@example.org after being assigned as scoped admin
- I am redirected to the admin T&C page
- I accept the terms
- at `http://localhost:3000/admin/admin_terms/accept`, I get `No route matches [PUT] "/admin/admin_terms/accept"`

File: app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb (hunk 2 above)
Comment (verarojman) [reply app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb:11]: Ahh, is it the scoped admins group you're creating here?

**Comment by verarojman** [comment]
Also:

- Having permissions to admin a specific assembly and
- Clicking on "edit" in the assemblies public page (`http://localhost:3000/assemblies/`)
- I click on an assembly that **I cannot edit**
- I visit (`http://localhost:3000/admin/assemblies/quo-beatae/edit`) and I get error `No route matches [GET] "/admin/assemblies/quo-beatae/edit"`

File: app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb (hunk 2 above)
Comment (microstudi) [reply app/commands/decidim/decidim_awesome/admin/create_scoped_admin.rb:11]: yep

**Comment by microstudi** [comment]
@verarojman I've made an clarification on how to test not-founds redirections in the description, basically, change the development.rb to:

```ruby
  # Show full error reports.
  config.consider_all_requests_local = false
```

**Pull request merged**